source/server/http/admin.cc

```diff
@@ -411,54 +411,67 @@ Http::Code AdminImpl::handlerStats(absl::string_view url, Http::HeaderMap& respo
                                    Buffer::Instance& response, AdminStream& admin_stream) {
   Http::Code rc = Http::Code::OK;
   const Http::Utility::QueryParams params = Http::Utility::parseQueryString(url);
+
+  const bool used_only = !(params.find("usedonly") == params.end());
+  const bool has_format = !(params.find("format") == params.end());
+
   std::map<std::string, uint64_t> all_stats;
   for (const Stats::CounterSharedPtr& counter : server_.stats().counters()) {
-    all_stats.emplace(counter->name(), counter->value());
+    const bool add_counter = used_only ? counter->used() : true;
+    if (add_counter) {
+      all_stats.emplace(counter->name(), counter->value());
+    }
   }
 
   for (const Stats::GaugeSharedPtr& gauge : server_.stats().gauges()) {
-    all_stats.emplace(gauge->name(), gauge->value());
+    const bool add_gauge = used_only ? gauge->used() : true;
+    if (add_gauge) {
+      all_stats.emplace(gauge->name(), gauge->value());
+    }
   }
 
   // TOOD(ramaraochavali): See the comment in ThreadLocalStoreImpl::histograms() for why we use a
   // multimap here. This makes sure that duplicate histograms get output. When shared storage is
   // implemented this can be switched back to a normal map.
   std::multimap<std::string, std::string> all_histograms;
   for (const Stats::ParentHistogramSharedPtr& histogram : server_.stats().histograms()) {
-    if (histogram->used()) {
-      std::vector<std::string> summary;
-      const std::vector<double>& supported_quantiles_ref =
-          histogram->intervalStatistics().supportedQuantiles();
-      summary.reserve(supported_quantiles_ref.size());
-      for (size_t i = 0; i < supported_quantiles_ref.size(); ++i) {
-        summary.push_back(fmt::format("P{}({},{})", 100 * supported_quantiles_ref[i],
-                                      histogram->intervalStatistics().computedQuantiles()[i],
-                                      histogram->cumulativeStatistics().computedQuantiles()[i]));
-      }
+    const bool add_histogram = used_only ? histogram->used() : true;
+    if (add_histogram) {
+      if (histogram->used()) {
+        std::vector<std::string> summary;
+        const std::vector<double>& supported_quantiles_ref =
+            histogram->intervalStatistics().supportedQuantiles();
+        summary.reserve(supported_quantiles_ref.size());
+        for (size_t i = 0; i < supported_quantiles_ref.size(); ++i) {
+          summary.push_back(fmt::format("P{}({},{})", 100 * supported_quantiles_ref[i],
+                                        histogram->intervalStatistics().computedQuantiles()[i],
+                                        histogram->cumulativeStatistics().computedQuantiles()[i]));
+        }
 
-      all_histograms.emplace(histogram->name(), absl::StrJoin(summary, " "));
+        all_histograms.emplace(histogram->name(), absl::StrJoin(summary, " "));
+      } else {
+        all_histograms.emplace(histogram->name(), "No recorded values");
```

Comment:
I forgot where we landed with this situation earlier, but what happens if there are two histograms of the same name?  Should we show values for both of them?

Reply:
yes